apis/ingress/v1alpha1/astartedefaultingress_types.go

```diff
@@ -0,0 +1,144 @@
+	// +optional
+	TLSSecret string `json:"tlsSecret,omitempty"`
+	// When true, enable Cross-Origin Resource Sharing (CORS). Default: false.
+	// +optional
+	Cors *bool `json:"cors,omitempty"`
+	// When true, the housekeeping endpoint is publicly exposed. Default: true.
+	// +optional
+	ExposeHousekeeping *bool `json:"exposeHousekeeping,omitempty"`
+	// When true, all /metrics endpoints for Astarte services will be served by the Ingress.
+	// Beware this might be a security hole. You can control which IPs can access /metrics
+	// with serveMetricsToSubnet. Default: false.
+	// +optional
+	ServeMetrics *bool `json:"serveMetrics,omitempty"`
+	// When specified and when serveMetrics is true, /metrics endpoints will be served only to IPs
+	// in the provided subnet range. The subnet has to be compatible with the HAProxy
+	// ACL src syntax (e.g.: "10.0.0.0/16"). Default: "".
+	// +optional
+	ServeMetricsToSubnet string `json:"serveMetricsToSubnet,omitempty"`
+}
+
+// AstarteDefaultIngressDashboardSpec defines how the Astarte Dashboard is served.
+type AstarteDefaultIngressDashboardSpec struct {
+	metav1.TypeMeta `json:",inline"`
+	// When true, deploy the Ingress for the Dashboard.
+	// +optional
+	Deploy *bool `json:"deploy,omitempty"`
+	// When true, enable TLS authentication for the Dashboard.
+	// +optional
+	SSL *bool `json:"ssl,omitempty"`
+	// The host handling requests addressed to the dashboard. When deploy is true and host is not set,
+	// the dashboard will be exposed at the following URL: https://<astarte-base-url>/dashboard.
+	// +optional
+	Host string `json:"host,omitempty"`
+	// The secret containing the TLS certificates and keys used to access the Astarte Dashboard. The secret
+	// must be present in the namespace in which Astarte resides.
+	// +optional
+	TLSSecret string `json:"tlsSecret,omitempty"`
+}
+
+// AstarteDefaultIngressBrokerSpec defines how the Astarte Broker is served.
+type AstarteDefaultIngressBrokerSpec struct {
+	metav1.TypeMeta `json:",inline"`
+	// When true, expose the Broker.
+	// +optional
+	Deploy *bool `json:"deploy,omitempty"`
+	// The secret containing the TLS certificates and keys used to connect to the Astarte Broker. The secret
+	// must be present in the namespace in which Astarte resides.
+	// +optional
+	TLSSecret string `json:"tlsSecret,omitempty"`
+	// Set the type of service employed to expose the broker. Supported values are "NodePort" and "LoadBalancer".
+	// Default: "LoadBalancer"
+	// +optional
+	ServiceType v1.ServiceType `json:"serviceType,omitempty"`
+	// Set the LoadBalancerIP if and only if the broker service is of type "LoadBalancer". This feature depends on
+	// wheather the cloud provider supports specifying the LoadBalancerIP when a load balancer is created.
+	// +optional
+	LoadBalancerIP string `json:"loadBalancerIP,omitempty"`
+}
+
+// AstarteDefaultIngressSpec defines the desired state of the AstarteDefaultIngress resource
+type AstarteDefaultIngressSpec struct {
```

Comment:
I think it gets cumbersome to have to fill the TLSSecret three times for API, Dashboard and Broker when in most deployments it will boil to a single certificate. I suggest having a TLSSecret variable in the main ingress spec, and then variables in the individual objects will override it in case they're specified